types/react/experimental.d.ts

```diff
@@ -38,7 +38,24 @@ import React = require('.');
 
 export {};
 
+type NonNullableValue = string | number | boolean | object | symbol;
+
 declare module '.' {
+    type MutableSource<TSource extends NonNullableValue, TVersion extends NonNullableValue> = {
+        _source: TSource,
+        _getVersion: (source: TSource) => TVersion,
+    };
+    type MutableSourceType<T> = T extends MutableSource<infer TSource, any> ? TSource : never;
```

Comment:
These `T` prefixes seem redundant. It is already a type. Since we also don't use `I` prefixes for interfaces I wouldn't want to start using `T` prefixes.